### PR TITLE
Add option to allow having an undercore in test methods

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
This addresses #251.

The first commit can be merged on its own via #255

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Ok, really don't get what th deal with this `ant-task.rst` is. I unstated it before my commit, after not making the change in the first place. Sometimes tooling can be such a pain.

**Comment by JeroenDeDauw** [comment]
@JanZerebecki I invoke your great regex wizard skills to enhance this PR and be closer to the rules described in https://github.com/phpmd/phpmd/issues/251 :)

Then I just need to find a git wizard as well to get rid of the `ant-task.rst` file of doom.

File: src/main/php/PHPMD/Rule/Controversial/CamelCaseMethodName.php [hunk 1]
```diff
@@ -101,4 +94,18 @@ public function apply(AbstractNode $node)
             }
         }
     }
+
+    private function isValid($methodName)
+    {
+        if ($this->getBooleanProperty('allow-underscore-test') && strpos($methodName, 'test') === 0) {
+            return preg_match('/^test[a-zA-Z0-9]*([_][a-z][a-zA-Z0-9]*)?$/', $methodName);
```
Comment (JanZerebecki) [inline src/main/php/PHPMD/Rule/Controversial/CamelCaseMethodName.php]: Modified to only allow:
    3 or more words (including "test") before the _
    2 or more words after the _
    or 1 or more words but no underscore
A word starts with an uper case A-Z, then zero or more lower case a-z, then zero or more numbers 0-9 are allowed.
/^test(([A-Z][a-z]*[0-9]*)+|([A-Z][a-z]*[0-9]*){3,}[_]([A-Z][a-z]*[0-9]*){2,}))$/
Tests not included. Wielding this regexp is unsafe without first properly testing it.

**Comment by JanZerebecki** [comment]
> Then I just need to find a git wizard as well to get rid of the ant-task.rst file of doom.

I don't get how this survived for so long. (The .gitattributes rules in this repository to convert text files from LF to the systems preferred line endings on checkout and internally to LF before diff/commit was added after this file was added; this file was added with CRLF, internally files are now stored with LF line endings.) The proper fix is to commit the converted ant-task.rst in a different commit, which I assume was the intention when adding the .gitattributes .

To get rid of it from this commit do:
$ git checkout HEAD~1 src/site/rst/documentation/ant-task.rst
$ git commit --amend

**Comment by JeroenDeDauw** [comment]
> /^test(([A-Z][a-z][0-9])+|([A-Z][a-z][0-9]){3,}_{2,}))$/

Thnx!

> To get rid of it from this commit do:
> $ git checkout HEAD~1 src/site/rst/documentation/ant-task.rst
> $ git commit --amend

Tried that and several other things. It always re-appears.

**Comment by JanZerebecki** [comment]
Yes it only gets rid of it from your commit. That it reappears as modified in your working copy is what the .gitattributes in the repo tells git to do. Also see above what I said about a proper fix.

**Comment by JeroenDeDauw** [comment]
@manuelpichler are you fine with this change? If so, I'll amend it with the latest regex

**Comment by addshore** [comment]
@manuelpichler @ravage84 any progress here?

**Pull request merged**